Invoke-ScriptInRemoteSessions.ps1

```diff
@@ -0,0 +1,91 @@
+Param (
+    [Parameter(Mandatory = $false)] [String] $ScriptFileName = "Clear-ComputeNode.ps1",
+    [Parameter(Mandatory = $false)] [String] $Addresses = "127.0.0.1",
+    [Parameter(Mandatory = $false)] [Switch] $IndividualCredentials,
+    [Parameter(Mandatory = $false, ValueFromRemainingArguments = $true)] $ArgumentsToPass
+)
+
+function New-Sessions {
+    Param (
+        [Parameter(Mandatory=$true)] [String[]] $Addresses,
+        [Parameter(Mandatory=$false)] [Int] $RetryCount = 10,
+        [Parameter(Mandatory=$false)] [Int] $Timeout = 300000
+    )
+
+    if ($IndividualCredentials -eq $false) {
+        $Credential = Get-Credential -Message "Enter common credentials for compute nodes"
+    }
+
+    Write-Host "Creating Powershell sessions..."
+    $Sessions = @()
+    foreach ($Address in $Addresses) {
+        try {
+            if ($IndividualCredentials -eq $true) {
+                $Credential = Get-Credential -Message "Enter credentials for $Address"
+            }
+            $PSO = New-PSSessionOption -MaxConnectionRetryCount $RetryCount -OperationTimeout $Timeout -ErrorAction Stop
+            $Session = New-PSSession -ComputerName $Address -Credential $Credential -SessionOption $PSO -ErrorAction Stop
+            $Sessions += $Session
+            Write-Host "Session created: $($Session.ComputerName)"
+        } catch {
+            Write-Host "ERROR: Failed to create session for remote compute node: $Address. Skipping this compute node."

```

Comment:
Maybe this should be a fatal error?

Reply:
Why? Let's run the script for valid sessions.